modular_zubbers/code/modules/storyteller/antagonist_event.dm

```diff
@@ -0,0 +1,170 @@
+/datum/round_event_control/antagonist
+	reoccurence_penalty_multiplier = 0
+	track = EVENT_TRACK_ROLESET
+	/// Protected roles from the antag roll. People will not get those roles if a config is enabled
+	var/list/protected_roles
+	/// Restricted roles from the antag roll
+	var/list/restricted_roles
+
+/datum/round_event_control/antagonist/New()
+	. = ..()
+	if(CONFIG_GET(flag/protect_roles_from_antagonist))
+		restricted_roles |= protected_roles
+
+/datum/round_event_control/antagonist/can_spawn_event(popchecks = TRUE)
+	. = ..()
+	if(!.)
+		return
+	if(!roundstart && !SSgamemode.can_inject_antags())
```

Comment:
Not sure myself, but shouldn't this be an or statement instead of an and statement?